### PR TITLE
docs: Further updates to quick-start

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs: Further updates to quick-start
Additional Description:

- code-block cleanups
- separate configs page/s
- add dynamic config info

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by phlax** [comment]
this is also still pretty wip (im testing between here and #13783) but docs are here

https://storage.googleapis.com/envoy-pr/13793/docs/start/quick-start/index.html

File: docs/BUILD [hunk 1]
```diff
@@ -17,6 +17,7 @@ filegroup(
             "root/**/*.pb",
         ],
         exclude = [
+            "root/**/envoy-dynamic*.yaml",
```
Comment (phlax) [inline docs/BUILD:20]: same comment as on other PR (https://github.com/envoyproxy/envoy/pull/13783#discussion_r514217718) - its not ideal to exclude these configs, but its not trivial to include either

File: docs/root/start/quick-start/configuration-dynamic-control-plane.rst [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+You may also wish to explore implementing your own control plane, in which case the
+`Go Control Plane <https://github.com/envoyproxy/go-control-plane>`_ provides a reference implementation
+that is a good place to start.
+
+At a minimum, you will need to start Envoy configured with the following sections:
+
+- :ref:`node <start_quick_start_dynamic_node>` to uniquely identify the proxy node.
+- :ref:`dynamic_resources <start_quick_start_dynamic_dynamic_resources>` to tell Envoy which configurations should be updated dynamically
+- :ref:`static_resources <start_quick_start_dynamic_static_resources>` to specify where Envoy should retrieve its configuration from.
+- :ref:`layered_runtime <start_quick_start_dynamic_layered_runtime>` to persist dynamically-provided configurations.
```
Comment (mattklein123) [inline docs/root/start/quick-start/configuration-dynamic-control-plane.rst:21]: This isn't actually required. Would it be simpler to just skip RTDS in the overview? WDYT?

File: docs/root/start/quick-start/configuration-dynamic-filesystem.rst [hunk 3]
```diff
@@ -0,0 +1,107 @@
+.. _start_quick_start_dynamic_filesystem:
+
+Configuration: Dynamic from filesystem
+======================================
+
+You can start Envoy with dynamic configuration that uses files as a form of
+:ref:`Extensible discovery service (xDS) <xds_protocol>`.
```
Comment (mattklein123) [inline docs/root/start/quick-start/configuration-dynamic-filesystem.rst:7]: nit: I think the "x" stands fro "*" effectively. I don't know that we actually have spelled out name for this, but I don't think it's extensible. :)

**Review by mattklein123 — CHANGES_REQUESTED** [review]
This is really awesome. Just a few small comments and we can ship and iterate!

/wait

File: docs/root/start/quick-start/configuration-dynamic-control-plane.rst (hunk 2 above)
Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-control-plane.rst:21]: in the sandbox it appeared to be required to make it work

im pro removing if its not absolutely necessary, but i would like to see it work - ill check this again

File: docs/root/start/quick-start/configuration-dynamic-filesystem.rst (hunk 3 above)
Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-filesystem.rst:7]: yep, so i searched on this to see if it was ever used previously, and no prior art  that i could find, but...

by convention, that is what a small `x` prefix in an acronym ~is (imhe anyway)

not sure "asteriskds" has the same ring to it 8/

Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-filesystem.rst:7]: i guess this is why i always read it that way...

https://en.wikipedia.org/wiki/XML

Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-filesystem.rst:7]: "x discovery service" so like something you have to pay for in a hotel room

Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-filesystem.rst:7]: to quote lewis carrol

"When I use a word," Humpty Dumpty said, in rather a scornful tone, "it means just what I choose it to mean—neither more nor less." "The question is," said Alice, "whether you can make words mean so many different things." "The question is," said Humpty Dumpty, "which is to be master—that's all."

Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-filesystem.rst:7]: i guess what im saying with my somewhat cryptic posts, is that we should define it (this is more of an issue with the glossary than here)

here, i think its not very helpful for a newcomer to read "...is a form of xds", if xds is not well-defined.

 As i still dont really understand what it means or stands for, i think its not well defined.

Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-filesystem.rst:7]: to be precise, not exactly no prior art https://www.envoyproxy.io/docs/envoy/latest/extending/extending#extending-envoy-for-custom-use-cases

File: docs/root/start/quick-start/configuration-dynamic-control-plane.rst (hunk 2 above)
Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-control-plane.rst]: i checked again, and it wasnt necessary, so i removed from both

File: docs/root/start/quick-start/configuration-dynamic-filesystem.rst (hunk 3 above)
Comment (phlax) [reply docs/root/start/quick-start/configuration-dynamic-filesystem.rst:7]: so, after going through docs/google a bit more and trying to think how to pin this down...

im still of the opinion that this term is used rather ambiguously, and that to have the level of involvement i now have and to be uncertain exactly what xDS is means something needs clarifying

im coming to the understanding/definition that perhaps xDS *is* xDS - ie the protocol name and is not exactly an acronym

ive updated the PR on that basis, and opened a glossary PR where i think we can pin down these terms.

**Review by mattklein123 — APPROVED** [review]
Nice!

**Pull request merged**